x/dymns/keeper/sell_order_name.go

```diff
@@ -0,0 +1,88 @@
+package keeper
+
+import (
+	errorsmod "cosmossdk.io/errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	dymnstypes "github.com/dymensionxyz/dymension/v3/x/dymns/types"
+	"github.com/dymensionxyz/gerr-cosmos/gerrc"
+)
+
+// CompleteDymNameSellOrder completes the active sell order of the Dym-Name,
+// give value to the previous owner, and transfer ownership to new owner.
+func (k Keeper) CompleteDymNameSellOrder(ctx sdk.Context, name string) error {
+	dymName := k.GetDymName(ctx, name)
+	if dymName == nil {
+		return errorsmod.Wrapf(gerrc.ErrNotFound, "Dym-Name: %s", name)
+	}
```

Comment:
why can this happen? seems like it should be impossible?

Reply:
This can happen, if someone jumped in and call existing method like they know it all.
And that's why I treat every method independently.

Btw: I have seen impossible things happened multiple times. Today it's not does not mean it won't happens tmr.